src/Lecture3.hs

```diff
@@ -213,9 +250,26 @@ types that can have such an instance.
 -- instance Foldable Weekday where
 -- instance Foldable Gold where
 -- instance Foldable Reward where
--- instance Foldable List1 where
--- instance Foldable Treasure where
-
+-- can't a non-polymorphic type implement Foldable in Haskell
+instance Foldable List1 where
+  foldr :: (a -> b -> b) -> b -> List1 a -> b
+  foldr func acc (List1 h []) = func h acc
+  foldr func acc (List1 h [t]) = func h (foldr func acc (List1 t []))
+  foldr func acc (List1 h (x : xs)) = foldr func (func x acc) (List1 h xs)
```

Comment:
That's a correct implementation 👍🏻 
You can simplify it by reusing the `Foldable` instance for ordinary lists 🚂 🚋 🚋 
The same is true for `foldMap`

Reply:
Updated